process/sp.py

```diff
@@ -277,14 +277,15 @@
         population_density,
         intersection_density,
     )
-        
+
     samplePointsData = samplePointsData[["hex_id", "edge_ogc_fid", "geometry"]].join(full_nodes, how="left")
-    
+
     # create binary distances evaluated against accessibility distance
+    # Options for distance decay accessibility scores are available in setup_sp.py module
     binary_names = [f"{x.replace('nearest_node','access')}_binary" for x in distance_names]
-    samplePointsData[binary_names] = (samplePointsData[distance_names] <= parameters['accessibility_distance']) \
-                                           .astype("Int64").fillna(0)
-    
+    samplePointsData[binary_names] = ssp.binary_access_score(samplePointsData, binary_names, distance_names, parameters['accessibility_distance'])

```

Comment:
Rather than bake the binary access score function into the code, perhaps we should make the choice user configurable in the setup (ie to address reviewer request)?  Then when the code gets here it checks for the configured formula type (perhaps binary is default), and then executes it along with any associated parameters.  For example, could be ifelse type logic responding to setup parameters.  What do you think?

Reply:
I thought of this too, but I just went for the easy route. I suppose users could just draw from the module for other methods, while keeping our workflow as is with the binary default. But I am open to the idea with ifelse option as well. Would you mind provide me with the code you have in mind? Thank you!

Reply:
Maybe what could work is if in the setup code there is a variable defined like
```
threshold_function = 'binary'
```
Then, in sp.py this is used like in the following (using a mock up samplePointsData frame):
```
# establish mock variables and parameters
samplePointsData = pd.DataFrame({'x1':[100,numpy.NaN,300,400,500,600,700,800,900,1000],'x2':[1000,900,800,700,600,500,400,300,200,100]})
distance_names = ['x1','x2'] 
access_names = [f'access_{x}' for x in distance_names]
parameters = {'accessibility_distance':500}

# create a dictionary of the available threshold functions -- 
# this could be done in setup_sp.py and called as ssp.threshold_functions[config[threshold_function]]
threshold_functions = {
    'binary':binary_access_score,
    'soft':soft_access_score,
    'cumulative':cumulative_gaussian_access_score
}
# define the access score variables
samplePointsData[access_names] = threshold_functions[threshold_function](samplePointsData, distance_names, parameters['accessibility_distance'])
```

in actual use, I think the line in sp.py would look like,
```
# define the access score variables
samplePointsData[access_names] = ssp.threshold_functions[config[threshold_function]](
    samplePointsData, distance_names, parameters['accessibility_distance']
)
```

Reply:
While I am working on this change, I felt like it seems more straightforward to just change the functions from sp.py, than  in the setup config code. 

It took me a while to understand the below:
```
ssp.threshold_functions[config[threshold_function]](
    samplePointsData, distance_names, parameters['accessibility_distance']
)
```

So I am thinking may be we should just keep it simple:
```
# create binary access scores evaluated against accessibility distance
# Options for distance decay accessibility scores are available in setup_sp.py module
access_score_names = [f"{x.replace('nearest_node','access')}_score" for x in distance_names]
samplePointsData[access_score_names] = ssp.binary_access_score(samplePointsData, distance_names, parameters['accessibility_distance'])
```

Our goal is to give users the alternatives of distance decay functions, but we still want to state that binary is the preferable option. I guess to keep the simple binary version in the main process would work for our advantages. Just want to share a thought, but I am open to the idea of config setup if anyone think that would be a better option!

Reply:
If you think that's best for submitting the article sooner than later, that's fine with me.  Alternately, if you thought it would be good I could do a commit with code for the setup next week (ie use of binary function as a default, matching our use case, but optional parameterisation to use the other functions external to the main code).  I understand if you want to keep things straightforward and lock in the changes now though.